packages/core/src/presets.ts

```diff
@@ -122,3 +122,18 @@ export const fly = () =>
     },
     runtimeEnv: process.env,
   });
+
+/**
+ * Cloudflare Pages Environment Variables
+ * @see https://developers.cloudflare.com/pages/configuration/build-configuration/#environment-variables
+ */
+export const cloudflarePages = () =>
+  createEnv({
+    server: {
+      CF_PAGES: z.string().optional(),
+      CF_PAGES_BRANCH: z.string().optional(),
+      CF_PAGES_COMMIT_SHA: z.string().optional(),
+      CF_PAGES_URL: z.string().optional(),
+    },
+    runtimeEnv: process.env,
```

Comment:
do we need a note that you need to enable node_compat to use this? I am not familiar with CF pages but at least wor workers envs are not available on global scope unless you opt-in to their node_compat flag, else you get it on request context